### ISSUE TITLE
https or SSL

### ISSUE THREAD

**Issue by teymourian** [ConsenSys-archive/ganache#3095]
Hi friends
I'm running program ganache
Can run ganache in https / SSl mode
Thanks

**Comment by davidmurdoch** [comment]
Not currently, no. What case did you have in mind for using ganache with an SSL certificate?

**Comment by teymourian** [comment]
HI @davidmurdoch 
When I use metamask wallet [on mobile] to test my network, using SSL is mandatory.
Thanks

**Comment by muratogat** [comment]
This would also be very, very useful to me.

I am also developing a mobile wallet that I need to test on local networks. iOS by default blocks any requests to insecure resources. Also when running my dapp in the in-app browser with node URL set to ganache, it rejects all requests by browser policy.

Currently doing all kinds of workarounds and dev-only settings (disabling iOS blocks, setting origin headers, etc.) to no avail. Thinking about running a reverse proxy to upgrade requests to SSL, but I'll need a domain for that and can't work with IP & port anymore.

Instead a simple --ssl flag would make things much easier. Angular does this somehow like "ng serve --ssl", even by creating a self-signed certificate when none is provided by the user. (https://angular.io/cli/serve)

Thanks

**Comment by MicaiahReid** [comment]
Thanks for the info, @muratogat! And thanks for the issue, @teymourian. 

We've bumped up the priority on this one because of the interest. However, as it stands it still may be some time before this is started. If either of you has any interest in getting this done sooner, a PR would be welcome!

**Comment by anthdm** [comment]
@muratogat  @teymourian You can use a let's encrypt certificate on top of a proxy Nginx server

**Comment by muratogat** [comment]
Unfortunately not.
Let's encrypt doesn't issue SSL certificate to IP addresses, only domains.
Mobile devices can't be modified to resolve these domains to your local IP's without maintaining your own DNS server.

**Comment by sahilrajput03** [comment]
> @muratogat @teymourian You can use a let's encrypt certificate on top of a proxy Nginx server

Hey, @anthdm your suggestion worked, thanks a lot.

I used free domain from https://www.noip.com/ to map to my dynamic IP address so its feasible for anyone out there like me but still I WOULD APPRECIATE A DIRECT WAY FROM GANACHE team TO MAKE IT POSSIBLE FOR SURE):

Dynamic DNS setting so it updates my IP address (I don't have a static IP address) on the no-ip server. no-ip server is free (you need to renew your domain on no-ip domain once in 21 days and its a bit tedious but they send you mail before the expiring date so they are pretty cool).

First of all I created a free domain and then added my login credentials of no-ip.com in my router's DNS settings so it updates whenever the broadband IP changes. Also, opened port 9544 on your desktop/wherever you're planning to host your ganache server.

![image](https://user-images.githubusercontent.com/31458531/196166525-0fab2022-4cfc-4d33-9ed7-823ebf9ce6f3.png)

My [`nginx.conf` file](https://github.com/sahilrajput03/nginx/blob/main/nginx.conf.TRUFFLE_BLOCKCHAIN_OVER_SSL): (I already created a SSL certificate using `certbot` cil, its free as well.)

![image](https://user-images.githubusercontent.com/31458531/196051643-10665512-a210-4ef8-b8be-f72c809c763e.png)

I used `ganache UI`  app to settings for the server and hit **restart** button on top-right:

![image](https://user-images.githubusercontent.com/31458531/196051703-70587a08-99ff-43f8-8533-79a7f1b210be.png)

Metamask Mobile App:

![image](https://user-images.githubusercontent.com/31458531/196051600-40119708-8170-42bd-b45b-d3c2c21b5e06.png)

So finally you should be able to see that page on URL like that:
![image](https://user-images.githubusercontent.com/31458531/196051679-57df3091-6058-4a63-b3d1-0b0798f995f9.png)

and in your mobile metamask app you can import the accounts using private key from the ganache-ui app and account in mobile would look like that:

![image](https://user-images.githubusercontent.com/31458531/196168507-56b4f95b-8618-4a1a-81f0-6bac97e032d5.png)


**SO BASICALLY NOW I OWN A PERSONAL BLOCKCHAIN testnet of my own.**
